python/pyarrow/tests/test_types.py

```diff
@@ -818,6 +818,19 @@ def test_fields_weakrefable():
     assert wr() is None
 
 
+def test_run_end_encoded_type():
+    ty = pa.run_end_encoded(pa.int64(), pa.utf8())
+    assert isinstance(ty, pa.RunEndEncodedType)
+    assert ty.run_end_type == pa.int64()
+    assert ty.value_type == pa.utf8()
+
+    with pytest.raises(TypeError):
+        pa.run_end_encoded(pa.int64(), None)
+
+    with pytest.raises(TypeError):
+        pa.run_end_encoded(None, pa.utf8())
```

Comment:
Can you also test with an unsupported run_end type (eg uint)

Reply:
Added. It raises a `ValueError` as the validation comes from the C++ side. Is that OK or should I write a Python validation that raise `TypeError`?

Reply:
ValueError is fine here I think (it's a bit ambiguous anyway, as you can argue that the "type" of the argument, i.e. a DataType is correct, but the exact value (int8 vs int16) is not correct. It's just that the values itself are also types ;))